### PR TITLE
fix(errors): fix loaders not throwing correctly

### PR DESCRIPTION
**Pull request by JamesSingleton**
Loaders were not properly throwing if index.js was not correctly formatted

### CODE REVIEW

File: packages/one-app-bundler/webpack/loaders/validate-one-app-compatibility-loader.js [hunk 1]
```diff
@@ -16,12 +16,12 @@ const loaderUtils = require('loader-utils');
 
 function validateOneAppCompatabilityLoader(content) {
   const { appCompatibility } = loaderUtils.getOptions(this);
-  const match = content.match(/export\s+default\s+(?!from)([\w\d]+)/);
+  const match = content.match(/export\s+default\s+(?!from)([\w\d]+;)/);
```
Comment (JAdshead) [inline packages/one-app-bundler/webpack/loaders/validate-one-app-compatibility-loader.js:19]: what about
```suggestion
  const match = content.match(/export\s+default\s+(?!from)([\w\d]+;)/).replace(';', '');
```

**Review by Francois-Esquire — COMMENTED** [review]
Would it be useful to abstract the same find and replace functionality for the three different loaders in one place? maybe use a callback to modify the export the way each needs to.

**Comment by Francois-Esquire** [comment]
To avoid `export default hoc(Component);` being used incorrectly by `hoc(Component).[META_DATA_KEY] = {...}` and assign to a separate instance, we should try using a named export `const holocronModule = ${match[1]};` and use the named variable to `export default holocronModule;`.
``

File: packages/one-app-bundler/webpack/loaders/validate-one-app-compatibility-loader.js (hunk 1 above)
Comment (JamesSingleton) [reply packages/one-app-bundler/webpack/loaders/validate-one-app-compatibility-loader.js:19]: @JAdshead that won't work as it gives a `Cannot read property 'replace' of null"`

File: packages/one-app-bundler/webpack/loaders/meta-data-loader.js [hunk 2]
```diff
@@ -17,12 +17,12 @@ const { META_DATA_KEY } = require('../..');
 
 function metaDataLoader(content) {
   const { packageJson: { version } } = readPkgUp.sync();
-  const match = content.match(/export\s+default\s+(?!from)([\w\d]+)/);
+  const match = content.match(/export\s+default\s+(?!from)([\w\d]+;)/);
```
[the remark this answers is not in the document]
Comment (JamesSingleton) [reply packages/one-app-bundler/webpack/loaders/meta-data-loader.js:20]: @Francois-Esquire 

Not entirely sure why, but this resulted in 
```
@americanexpress/one-app-bundler: Module must use `export default VariableName` in index syntax to use requiredExternals
```

For something like `export default ModuleContainer;` which should of passed. Logging it out resulted `null`

[the remark this answers is not in the document]
Comment (Francois-Esquire) [reply packages/one-app-bundler/webpack/loaders/meta-data-loader.js:20]: Strange...

**Review by Francois-Esquire — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

**Pull request merged**